### ISSUE TITLE
OneSignalStateSynchronizer risen ConcurrentModificationException

### ISSUE THREAD

**Issue by dekrandroid** [OneSignal/OneSignal-Android-SDK#73]
Sync failed

````VB
Fatal Exception: java.util.ConcurrentModificationException
       at java.util.LinkedHashMap$LinkedHashIterator.nextEntry(LinkedHashMap.java:346)
       at java.util.LinkedHashMap$KeyIterator.next(LinkedHashMap.java:366)
       at com.onesignal.OneSignalStateSynchronizer.generateJsonDiff(OneSignalStateSynchronizer.java:74)
       at com.onesignal.OneSignalStateSynchronizer.access$200(OneSignalStateSynchronizer.java:46)
       at com.onesignal.OneSignalStateSynchronizer$UserState.generateJsonDiff(OneSignalStateSynchronizer.java:263)
       at com.onesignal.OneSignalStateSynchronizer$UserState.access$000(OneSignalStateSynchronizer.java:197)
       at com.onesignal.OneSignalStateSynchronizer.syncUserState(OneSignalStateSynchronizer.java:476)
       at com.onesignal.OneSignalStateSynchronizer$NetworkHandlerThread$1.run(OneSignalStateSynchronizer.java:440)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:158)
       at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by jkasten2** [comment]
@dekrandroid Could you let us know what version of the SDK your using? Also are you able to reproduce the issue? Lastly could you let me know what SDK methods you are calling in your app?

Thanks.

**Comment by jkasten2** [comment]
@dekrandroid I examined the stack trace closer and it the line number match up with our [2.4.1 SDK version](https://github.com/OneSignal/OneSignal-Android-SDK/blob/2.4.1/OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalStateSynchronizer.java). We fixed this issue in our [2.4.2 SDK](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/2.4.2). However we recommend updating to the latest 2.6.0 by using using the following in your .gradle if if you are not already.
`compile 'com.onesignal:OneSignal:2.+@aar'`
If you have this you will need to resync your gradle file to get the last version.

Feel free to reopen this issue if you see this issue on the latest version as well.
